Xamarin.Forms.Platform.Android/Renderers/LabelRenderer.cs

```diff
@@ -197,6 +202,15 @@ void UpdateLineHeight()
 				_view.SetLineSpacing(0, (float)Element.LineHeight);
 		}
 
+		void UpdateMaxLines()
+		{
+			if (Element.MaxLines > 0)
+			{
+				Control.SetSingleLine(Element.MaxLines == 1);
+				Control.SetMaxLines(Element.MaxLines);
+			}
```

Comment:
a `else` is missing. it should handled the case where a `MaxLines` value is set, then `ClearValue()`'d, actually reverting to the control default.